contribute/pull-request.md

```diff
@@ -0,0 +1,94 @@
+# Create a pull request
+
+We're excited that you're considering making a contribution to the Grafana project! This document guides you through the process of creating a [pull request](https://help.github.com/en/articles/about-pull-requests/).
+
+## Before you begin
+
+We know you're excited to create your first pull request. Before we get started, read these resources first:
+
+- Learn how to start [Contributing to Grafana](https://grafana.com/docs/contribute/overview/).
+- Make sure your code follows the relevant [style guides](/contribute/style-guides).
+
+## Your first pull request
+
+If this is your first time contributing to a open-source project on GitHub, make sure you read about [Creating a pull request](https://help.github.com/en/articles/creating-a-pull-request).
+
+To increase the chance of having your pull request accepted, make sure your pull request follows these guidelines:
+
+- Title and description matches the implementation.
+- Commits within the pull request follow the [style guide](#style-guide).
+- The pull request closes one related issue.
+- The pull request contains necessary tests that verify the intended behavior.
+- If your pull request has conflicts, rebase your branch onto the master branch.
+
+If the pull request fixes a bug:
+
+- The pull request description must include `Closes #<issue number>` or `Fixes #<issue number>`.
+- To avoid regressions, the pull request should include tests that replicate the fixed bug.
+
+### Frontend-specific guidelines
+
+Pull requests for frontend contributions must:
+
+- Use [Emotion](https://github.com/grafana/grafana/blob/master/style_guides/styling.md) for styling.
+- Not increase the Angular code base.
+- Not use `any` or `{}` without comments describing why.
+- Not contain large React components that could easily be split into several smaller components.
+- Not contain backend calls directly from components—use actions and Redux instead.
```

Comment:
I get the feeling these two might be _very_ subjective and open to interpretation - and to a certain point condescending for the author. I would be in favour of removing them or moving them into a front-end specific guideline document.

Reply:
I wasn't sure about the frontend-specific guidelines, but I think we should move as many of these as we can into the corresponding style guide. Maybe @dprokop has a suggestion?